File: Problems/LeetCode/easy/37-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        if len(s) != len(t): return False

        dic = {}
        for i in range(0, len(s)):
            if s[i] in dic: dic[s[i]] += 1
            else: dic[s[i]] = 1
            if t[i] in dic: dic[t[i]] -= 1
            else: dic[t[i]] = -1
        for v in dic.values():
            if v > 0: return False
        return True

instance = Solution()

print(instance.isAnagram("a", "ab"))

File: Problems/LeetCode/easy/37-valid-anagram.js
/*

Given two strings s and t, write a function to determine if t is an anagram of s.

Example 1:
    Input: s = "anagram", t = "nagaram"
    Output: true

Example 2:
    Input: s = "rat", t = "car"
    Output: false
Note:
You may assume the string contains only lowercase alphabets.

Follow up:
What if the inputs contain unicode characters? How would you adapt your solution to such case?

*/

let isAnagram = function(s, t) {
    
    if (s.length !== t.length) return false;

    const set = {};
    for (let i = 0; i < s.length; i++) {
        if (!set[s[i]]) set[s[i]] = 1;
        else set[s[i]]++;
        if (!set[t[i]])  set[t[i]] = -1;
        else set[t[i]]--;
    }
    
    return Object.values(set).every(v => v === 0); 
};

console.log('Example 1: Result - ' + isAnagram("anagram", "nagaram") + ' Expected: true');
console.log('Example 1: Result - ' + isAnagram("rat", "car") + ' Expected: false');